modules/videoio/include/opencv2/videoio.hpp

```diff
@@ -211,6 +211,7 @@ enum VideoCaptureProperties {
        CAP_PROP_CODEC_EXTRADATA_INDEX = 68, //!< Positive index indicates that returning extra data is supported by the video back end.  This can be retrieved as cap.retrieve(data, <returned index>).  E.g. When reading from a h264 encoded RTSP stream, the FFmpeg backend could return the SPS and/or PPS if available (if sent in reply to a DESCRIBE request), from calls to cap.retrieve(data, <returned index>).
        CAP_PROP_FRAME_TYPE = 69, //!< (read-only) FFmpeg back-end only - Frame type ascii code (73 = 'I', 80 = 'P', 66 = 'B' or 63 = '?' if unknown) of the most recently read frame.
        CAP_PROP_N_THREADS = 70, //!< (**open-only**) Set the maximum number of threads to use. Use 0 to use as many threads as CPU cores (applicable for FFmpeg back-end only).
+       CAP_PROP_CODEC_FOURCC = 71, //!< 4-character code representing type of the encoded data (the codec used).  This differs from CAP_PROP_FOURCC which can represent additional  information about the codec. e.g. CAP_PROP_FOURCC can return divx, xvid, mp4v etc. for an MPEG-4 video where CAP_PROP_CODEC_FOURCC would return only FMP4 (applicable for FFmpeg back-end only).
```

Comment:
There is some mess with:

```
CAP_PROP_FOURCC         =6, //!< 4-character code of codec. see VideoWriter::fourcc .
```

Both describes the "codec".
We could deprecate the old one, but we need robust and clear solution.

BTW, there are 3 different FOURCC could be used in some pipelines: camera firmware/device level(MJPEG compression), codec(YUV), output of VideoCapture(BGR24).

Reply:
@opencv-alalek updated to translate `codec_id` to fourcc and fallback to `codec_tag` if this doesn't exist.  This seems like a robust solution, the returned value will now "always" (apart from when the `codec_id` field doesn't exist) be the `4-character code of codec`.